api/app/controllers/webhook.controller.ts

```diff
@@ -154,3 +165,178 @@ export async function create(req: Request, res: Response) {
     }
   }
 }
+
+async function processCommit(
+  commit: any,
+  pushEvent: any,
+  type: "added" | "modified"
+) {
+  const changedFiles = [...commit[type]]; // get the files that were added or modified in the commit so we don't have to maintain two separate functions, this is also easily extendable when we want to take care of deleted commits.
+  for (const file of changedFiles) {
+    const rawUrl = `https://raw.githubusercontent.com/${pushEvent.repository.full_name}/master/${file}`;
+    const response: any = await axios.get(rawUrl);
+    const mdContent = response.data;
+    const jsonContent: BaseParsedMdContent =
+      parseMdToJSON<BaseParsedMdContent>(mdContent);
+    const transcript_by = jsonContent.transcript_by.toLowerCase();
+
+    function isTranscriptValid(transcript_by: string): boolean {
+      return (
+        transcript_by.includes("tstbtc") &&
+        transcript_by.includes("--needs-review")
+      );
+    }
+
+    if (!jsonContent) {
+      throw new Error(
+        "Malformed data: transcript content might not be in the correct format"
+      );
+    }
+
+    const transcriptHash = generateUniqueHash(jsonContent);
+    const totalWords = getTotalWords(jsonContent.body);
+    const content = jsonContent;
+
+    if (!transcriptHash || !totalWords) {
+      throw new Error(
+        "Malformed data: transcript content might not be in the correct format"
+      );
+    }
+
+    const existingTranscript = await Transcript.findOne({
+      where: { transcriptHash: transcriptHash },
+    });
+
+    if (existingTranscript && type === "added") {
+      throw new Error("transcript already exists");
+    }
+
+    if (!isTranscriptValid(transcript_by)) {
+      throw new Error(
+        "Transcript not from TSTBTC or does not need review - did not queue transcript"
+      );
+    }
+
+    const transcript: TranscriptAttributes = {
+      originalContent: {
+        ...content,
+        title: content.title.trim(),
+      },
+      content: content,
+      transcriptHash,
+      transcriptUrl: rawUrl,
+      status: TranscriptStatus.queued,
+      contentTotalWords: totalWords,
+    };
+
+    let transcriptData: any;
+    if (type === "added") {
+      transcriptData = await Transcript.create(transcript);
+    } else if (type === "modified") {
+      // Find the existing transcript in the database
+      const existingTranscript = await Transcript.findOne({
+        where: { transcriptUrl: rawUrl },
+      });
+
+      if (!existingTranscript) {
+        throw new Error("No transcript found to update");
```

Comment:
For cases where the file is modified and there is no corresponding queued transcript, I think its fine to create a new transcript and queue it.

Why?
Because this step runs after the `--needs-review` flag and `TSTBTC` check. So we are sure it's a transcript that well `needs a review`.

This is important for when we want to a soft phase out of previously queued transcript, all we need to do is delete them from the queue and then we edit the transcript in the repo to `--need-review` rather than deleting the file from the repo and creating it anew.

Reply:
This is still unresolved @nully0x